### ISSUE TITLE
Operator "==" not supported for types Type[]

### ISSUE THREAD

**Issue by ytimenkov** [microsoft/pylance-release#705]
## Environment data
-   Language Server version: Pylance language server 2020.12.0 (pyright ea581b3f)
-   OS and version: Ubuntu 20.20
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.5 venv

## Expected behaviour

I'm writing default exception handler and there I need to compare exception types:
```python
import json
import sys
from types import TracebackType
from typing import Type, cast


def exception_handler(
    etype: Type[BaseException], exc: BaseException, tb: TracebackType
):
    if etype == json.decoder.JSONDecodeError:
        exc = cast(json.decoder.JSONDecodeError, exc)
        print(f"Invalid json:\n{exc.doc!r}")


sys.excepthook = exception_handler

json.loads("invalid")
```

## Actual behaviour

Pylance complains:
```
Operator "==" not supported for types "Type[BaseException]" and "Type[JSONDecodeError]"
  Operator "==" not supported for types "Type[BaseException]" and "Type[JSONDecodeError]"Pylance (reportGeneralTypeIssues)
```

However running prints:
```
Invalid json:
'invalid'
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.12.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020121-9-december-2020
